### PR TITLE
[kata_containers] Add new plugin for Kata Containers

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a new plugin for the Kata Containers project runtime
(`kata-runtime`).

See:

  - https://github.com/kata-containers
  - https://katacontainers.io/

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

**Comment by jodh-intel** [comment]
Example output generated with:

```
$ sudo python ./sosreport -a -o kata_containers
```
[sos-report-kata-containers.zip](https://github.com/sosreport/sos/files/2012417/sos-report-kata-containers.zip)

File: sos/plugins/kata_containers.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+class KataContainers(Plugin, RedHatPlugin, DebianPlugin,
+                     UbuntuPlugin, SuSEPlugin):
+    """Kata Containers configuration
+    """
+
+    plugin_name = 'kata_containers'
+    profiles = ('system', 'virt')
+
+    runtime = 'kata-runtime'
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply sos/plugins/kata_containers.py]: Fixed.

File: sos/plugins/kata_containers.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    # trigger this plugin if the following package is installed
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply sos/plugins/kata_containers.py]: Fixed.

File: sos/plugins/kata_containers.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        # obtain a list of config files by asking the runtime
+        cmd = '{} --kata-show-default-config-paths'.format(self.runtime)
+        configs = self.get_command_output(cmd)
+        if configs and configs['status']:
+            for config in configs['output'].splitlines():
+                if config != "":
+                    config_files.append(config)
+
+        # get a unique list of config files
+        config_files = set(config_files)
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply sos/plugins/kata_containers.py]: Fixed.

File: sos/plugins/kata_containers.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        self.add_copy_spec(config_files)
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply sos/plugins/kata_containers.py]: Fixed.

File: sos/plugins/kata_containers.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    def setup(self):
+        self.limit = self.get_option("log_size")
+
+        if self.get_option("all_logs"):
+            # no limit on amount of data recorded
+            self.limit = None
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply sos/plugins/kata_containers.py]: I'm not sure I'm clear what I need to do here?

File: sos/plugins/kata_containers.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+        self.add_cmd_output('{} kata-env'.format(self.runtime))
+        self.attach_kata_config_files()
```
[the remark this answers is not in the document]
Comment (jodh-intel) [reply sos/plugins/kata_containers.py]: Fixed.

File: sos/plugins/kata_containers.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (bmr-cymru) [reply sos/plugins/kata_containers.py]: For now you can just delete lines 29-34 - the only logs you seem to collect are journal logs, and we don't currently support size limiting those.

When the `add_journal()` size limiting lands, you can either add the `sizelimit=` parameter to those calls (using `get_option("log_size")`), or (what I'd recommend), just do nothing and wait for the automatic size limiting to land - these are both in the review queue at the moment and we want to get them in in time for 3.6 (next month or so).

**Comment by jodh-intel** [comment]
Thanks for the review @bmr-cymru. Branch updated.

**Comment by jodh-intel** [comment]
Updated for new `container` profile.

**Comment by bmr-cymru** [comment]
Thanks! I'd been adding these as they were merged for plugins that landed after @TurboTurtle 's patch. Appreciate the time-save here.

**Pull request closed without merging**